build/components.ts

```diff
@@ -81,15 +69,15 @@ async function buildComponents() {
       const entry = path.resolve(p, './index.ts')
       if (!fs.existsSync(entry)) return
 
-      const external = (id) => {
-        return (
-          id.startsWith(VUE_REGEX) ||
-          id.startsWith(VUE_MONO) ||
-          id.startsWith(EP_PREFIX) ||
-          externals.some((i) => id.startsWith(i))
-        )
+      const rollupConfig = {
+        input: entry,
+        plugins,
+        external: getExternals({ full: false }),
       }
-      const esm = {
+      const bundle = await rollup.rollup(rollupConfig)
+
+      // ESM
+      await bundle.write({
```

Comment:
Actually I was wondering if we can put `Promise.all(writers)` here to enable a parallel writing(hypothetically, I am not very familiar with NodeJS stuff. cc @adaex is that possible?)

Reply:
I think it works.

Reply:
Maybe I will try to improve it in next PR. Also, we can execute multiple commands in parallel in build.sh

Reply:
> Actually I was wondering if we can put `Promise.all(writers)` here to enable a parallel writing

Yeah, it's perfectly fine to write to both `es` and `cjs` formats in parallel here.

Reply:
Let's move this into separate PRs for faster build speed.